src/main/java/Main.java

```diff
@@ -0,0 +1,174 @@
+
+        SessionFixBean sessionFixBean = new SessionFixBean();
+        sessionFixBean.createConfig();    //create config file for FIX
+        String sessionAlias = new SessionFixBean().getSessionAlias();
+
+        ConnectionID connectionID = ConnectionID.newBuilder().setSessionAlias(sessionAlias).build();
+
+        Event curEvent = MessageRouterUtils.storeEvent(eventRouter, Event.start(), null);
+        curEvent.name("FIX client " + sessionAlias + Instant.now());
+        curEvent.type("Microservice");
+        String rootEventId = curEvent.getId();
+
+        FixClient fixClient = new FixClient(new SessionSettings("src/main/resources/acceptor/acceptor.cfg"),
+                messageRouter, eventRouter, connectionID, rootEventId);
+
+        resources.add(new Resources("client", fixClient::stop));
+
+        ClientController controller = new ClientController(fixClient);
+
+        MessageListener<MessageGroupBatch> listener = (consumerTag, groupBatch) -> {
+            if (!controller.isRunning) controller.start(settings.autoStopAfter);
+
+            groupBatch.getGroupsList().forEach((group) -> {
+                try {
+                    if (group.getMessagesCount() != 1) {
+                        log.error("Message group contains more than 1 message", new IllegalArgumentException());
+                        throw new IllegalArgumentException();
+                    }
+                    AnyMessage message = group.getMessagesList().get(0);
+                    if (!message.hasRawMessage()) {
+                        log.error("Message in the group is not a raw message", new IllegalArgumentException());
+                        throw new IllegalArgumentException();
+                    }
+
+                    Session session = fixClient.getApplication().getSession();
+                    Message fixMessage = MessageUtils.parse(session, new String(message.getRawMessage().getBody().toByteArray(), StandardCharsets.UTF_8));
+                    session.send(fixMessage);
+
+                } catch (Exception e) {
+                    try {
+                        log.error("Failed to handle message group: " + MessageUtil.toPrettyString(group), e);
+                        MessageRouterUtils.storeEvent(eventRouter, rootEventId, "Failed to handle message group: " + MessageUtil.toPrettyString(group), "Error", e);
+                    } catch (InvalidProtocolBufferException invalidProtocolBufferException) {
+                        log.error("Failed to handle message group: ", invalidProtocolBufferException);
+                    }
+                }
+            });
+        };
+
+        try {
+            SubscriberMonitor monitor = messageRouter.subscribe(listener, INPUT_QUEUE_ATTRIBUTE);
+            if (monitor != null) {
+                resources.add(new Resources("raw-monitor", monitor::unsubscribe));
+            }
+        } catch (Exception e) {
+            log.error("Failed to subscribe to input queue", e);
```

Comment:
Should be an exception since we shouldn't proceed running if we cannot subscribe